packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -117,3 +165,24 @@ export function cloudflare<T extends Record<string, WorkerOptions>>(
 		},
 	};
 }
+
+/**
+ * Returns a set of environment names that need Node.js compatibility.
+ */
+function getNodeCompatEnvironments(config: PluginConfig, root: string) {
+	const needsNodeJsCompat = new Set<string>();
+
+	Object.entries(config.workers).forEach(([name, rawWorkerOptions]) => {
+		const wranglerConfigPath = path.resolve(
+			root,
+			rawWorkerOptions.wranglerConfig ?? './wrangler.toml',
+		);
+		const { workerOptions } =
+			unstable_getMiniflareWorkerOptions(wranglerConfigPath);
+		if (isNodeCompat(workerOptions)) {
+			needsNodeJsCompat.add(name);
+		}
+	});
+
+	return needsNodeJsCompat;
+}
```

Comment:
I thought we already restart the server when the config changes? Perhaps I was thinking of something else.

Reply:
The Vite server automatically restarts if the Vite config changes. In this case it would be the Wrangler config that changes. Usually, if the Wrangler config changes then we call `miniflare.setOptions` but in this case we would want to restart Vite I think.

Reply:
I refactored this stuff here https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/49/commits/6756ed24b3c690d3d78279bf782323f2112ac26b

But I agree we need to re-run the config if the wrangler.toml node-compat status changes...
How easy is it to reload config from within Vite?

Reply:
I think you would just call `viteDevServer.restart()`.